### PR TITLE
RT-649 Wallet Context

### PR DESCRIPTION
**Pull request by iandjx**
## Description
1. Added walletcontext package
2. Added example react js implementation of walletcontext

## To test
1. cd to /examples 
2. `npm i && npm  run dev`

### CODE REVIEW

**Comment by iandjx** [comment]
error
```
Unhandled Runtime Error
Error: Text content does not match server-rendered HTML.

See more info here: https://nextjs.org/docs/messages/react-hydration-error
```
appears when selected wallet is injected and is stored in local storage

**Comment by iandjx** [comment]
> error
> 
> ```
> Unhandled Runtime Error
> Error: Text content does not match server-rendered HTML.
> 
> See more info here: https://nextjs.org/docs/messages/react-hydration-error
> ```
> 
> appears when selected wallet is injected and is stored in local storage

resolved

**Comment by iandjx** [comment]
putting this back in draft. i think i figure out how to use web3react provider properly

File: WalletContext/pnpm-lock.yaml [hunk 1]
```diff
@@ -0,0 +1,6460 @@
+lockfileVersion: 5.4
+
```
Comment (MarkCBall) [inline WalletContext/pnpm-lock.yaml:2]: we are using yarn everywhere, no?

Comment (iandjx) [reply WalletContext/pnpm-lock.yaml:2]: yes, sorry, will take this out in favor of yarn

File: WalletContext/src/context/WalletProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
+import { Web3ReactHooks, Web3ReactProvider } from '@web3-react/core'
```
[the remark this answers is not in the document]
Comment (iandjx) [reply WalletContext/src/context/WalletProvider.tsx:1]: the existing wallet context will deprecated in favor of having individual wallet contexts per iframe. i've imported network and token constants to this project. for every new protocol we add, the constants of this project will be updated

File: WalletContext/src/chains.ts [hunk 3]
```diff
@@ -0,0 +1,204 @@
+import type { AddEthereumChainParameter } from '@web3-react/types'
+
+const INFURA_KEY = process.env.REACT_APP_INFURA_KEY
+  ? process.env.REACT_APP_INFURA_KEY
+  : process.env.NEXT_PUBLIC_APP_INFURA_KEY
+
+const ALCHEMY_KEY = process.env.REACT_APP_ALCHEMY_KEY
```
Comment (MarkCBall) [inline WalletContext/src/chains.ts:7]: would these be better as props? it seems like a bit of a bad pattern to expect that whatever imports this library now needs to set these exact ENV vars. 

If not, this expectation should be make very clear. Docs., env.example ect

File: WalletContext/pnpm-lock.yaml (hunk 1 above)
Comment (MarkCBall) [reply WalletContext/pnpm-lock.yaml:2]: i hate to be super annoying, but ping here =p

**Review by MarkCBall — APPROVED** [review]

File: WalletContext/src/chains.ts (hunk 3 above)
Comment (iandjx) [reply WalletContext/src/chains.ts:7]: image

ive tried passing the env variables as props but it ended up throwing 401 errors. 

will just make it clear in docs and .env example that infura and alchemy key needs to be passed

**Pull request merged**